File: AESEncryptUtil.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace com.daervin.common.utils
{
    public class AESEncryptUtil
    {
        private static String KEY = "7dub@utilAESDEnc";

        /// <summary>
        /// AES加密
        /// </summary>
        /// <param name="data">原始串</param>
        /// <returns>加密后的串</returns>
        public static string aesEncrypt(String data)
        {
            if (string.IsNullOrEmpty(data))
            {
                return null;
            }
            Byte[] toEncryptArray = Encoding.UTF8.GetBytes(data);
            Byte[] keyBytes = new Byte[16];
            Array.Copy(Encoding.ASCII.GetBytes(KEY), keyBytes,16);
            System.Security.Cryptography.RijndaelManaged rm = new System.Security.Cryptography.RijndaelManaged()           
            {
                Key = keyBytes,
                Mode = System.Security.Cryptography.CipherMode.ECB,
                Padding = System.Security.Cryptography.PaddingMode.PKCS7
            };

            System.Security.Cryptography.ICryptoTransform cTransform = rm.CreateEncryptor();
            Byte[] resultArray = cTransform.TransformFinalBlock(toEncryptArray, 0, toEncryptArray.Length);

            return byteToHexStr(resultArray);
        }

        /// <summary>
        /// AES解密
        /// </summary>
        /// <param name="data">加密后的串</param>
        /// <returns>原始串</returns>
        public static string aesDecrypt(String data)
        {
            if (string.IsNullOrEmpty(data))
            {
                return null;
            }

            byte[] content = hex2Bytes(data);
            Byte[] keyBytes = new Byte[16];
            Array.Copy(Encoding.ASCII.GetBytes(KEY), keyBytes, 16);

            System.Security.Cryptography.RijndaelManaged rm = new System.Security.Cryptography.RijndaelManaged
            {
                Key = keyBytes,
                Mode = System.Security.Cryptography.CipherMode.ECB,
                Padding = System.Security.Cryptography.PaddingMode.PKCS7
            };

            System.Security.Cryptography.ICryptoTransform cTransform = rm.CreateDecryptor();
            Byte[] resultArray = cTransform.TransformFinalBlock(content, 0, content.Length);
            return Encoding.UTF8.GetString(resultArray);
        }

        /// <summary>
        /// 字节数组转16进制字符串
        /// </summary>
        /// <param name="bytes">字节数组</param>
        /// <returns>16进制字符串</returns>
        private static string byteToHexStr(byte[] bytes)
        {
            string returnStr = "";
            if (bytes != null)
            {
                for (int i = 0; i < bytes.Length; i++)
                {
                    returnStr += bytes[i].ToString("x2");
                }
            }
            return returnStr;
        }

        /// <summary>
        /// 16进制字符串转字节数组
        /// </summary>
        /// <param name="hexStr">16进制字符串</param>
        /// <returns>字节数组</returns>
        private static byte[] hex2Bytes(String hexStr)
        {
            hexStr = hexStr.Replace(" ", "");
            if ((hexStr.Length % 2) != 0)
                hexStr += " ";
            byte[] returnBytes = new byte[hexStr.Length / 2];
            for (int i = 0; i < returnBytes.Length; i++)
                returnBytes[i] = Convert.ToByte(hexStr.Substring(i * 2, 2), 16);
            return returnBytes;
        }


        internal static string aesEncrypt(long p)
        {
            throw new NotImplementedException();
        }

        internal static string aesEncrypt(Func<string> func)
        {
            throw new NotImplementedException();
        }

     
    }
}

File: AESEncryptUtil.java
package com.dub.common.utils;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import javax.crypto.Cipher;
import javax.crypto.spec.SecretKeySpec;
import java.util.Arrays;

/**
 * Created by 7dub on 2017/4/19.
 */
public final class AESEncryptUtil {
    private static final Logger LOGGER = LoggerFactory.getLogger(AESEncryptUtil.class);
    private static final String KEY = "7dub@utilAESDEnc";

    private AESEncryptUtil() {
        // 不可实例化
    }

    /**
     * 加密
     *
     * @param data 待加密字符串
     * @return 加密字符串
     */
    public static String aesEncrypt(String data) {
        if (null == data) {
            return null;
        }
        try {
            byte[] keyBytes = Arrays.copyOf(KEY.getBytes("ASCII"), 16);
            SecretKeySpec skey = new SecretKeySpec(keyBytes, "AES");
            Cipher cipher = Cipher.getInstance("AES/ECB/PKCS5Padding");
            cipher.init(Cipher.ENCRYPT_MODE, skey);
            return bytes2Hex(cipher.doFinal(data.getBytes("utf-8")));
        } catch (Exception e) {
            LOGGER.warn("aesEncrypt fail", e);
            return null;
        }
    }

    public static String bytes2Hex(byte[] data) {
        StringBuilder buffer = new StringBuilder();
        for (int i = 0; i < data.length; i++) {
            String hexData = Integer.toHexString(data[i] & 0XFF);
            if (hexData.length() < 2) {
                buffer.append("0").append(hexData);
            } else {
                buffer.append(hexData);
            }
        }

        return buffer.toString();
    }

    public static byte[] hex2Bytes(String hexStr) {
        if (hexStr.length() < 1)
            return new byte[0];
        byte[] result = new byte[hexStr.length() / 2];
        for (int i = 0; i < hexStr.length() / 2; i++) {
            int high = Integer.parseInt(hexStr.substring(i * 2, i * 2 + 1), 16);
            int low = Integer.parseInt(hexStr.substring(i * 2 + 1, i * 2 + 2),
                    16);
            result[i] = (byte) (high * 16 + low);
        }
        return result;
    }

    /**
     * 解密
     *
     * @param data 加密字符串
     * @return 原始内容
     */
    public static String aesDecrypt(String data) {
        if (null == data) {
            return null;
        }
        try {
            byte[] content = hex2Bytes(data);
            byte[] keyBytes = Arrays.copyOf(KEY.getBytes("ASCII"), 16);
            SecretKeySpec skey = new SecretKeySpec(keyBytes, "AES");
            Cipher cipher = Cipher.getInstance("AES/ECB/PKCS5Padding");
            cipher.init(Cipher.DECRYPT_MODE, skey);
            byte[] result = cipher.doFinal(content);
            return new String(result, "utf-8");
        } catch (Exception e) {
            LOGGER.warn("aesDecrypt fail", e);
            return null;
        }
    }

    public static void main(String[] args) {
        String s = "123456";
        String sc = aesEncrypt(s);
        System.out.println("加密：" + sc);
        System.out.println("解密：" + aesDecrypt(sc));
    }
}
